app/src/test/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegrationTests.java

```diff
@@ -0,0 +1,271 @@
+//        AppsflyerIntegration.deepLinkListener = mock(AppsflyerIntegration.ExternalDeepLinkListener.class);
+//
+//        Integration<AppsFlyerLib> integration =
+//                (Integration<AppsFlyerLib>) AppsflyerIntegration.FACTORY.create(settings,analytics);
+//        verify(appsFlyerLib).setDebugLog(logger.logLevel!=Analytics.LogLevel.NONE);
+//        ArgumentCaptor<AppsflyerIntegration.ConversionListener> captorListener = ArgumentCaptor.forClass(AppsflyerIntegration.ConversionListener.class);
+//        ArgumentCaptor<String> captorDevKey = ArgumentCaptor.forClass(String.class);
+//        ArgumentCaptor<Context> captorContext = ArgumentCaptor.forClass(Context.class);
+//        verify(appsFlyerLib).init(captorDevKey.capture(), captorListener.capture() , captorContext.capture());
+//        Assert.assertTrue(captorListener.getValue()!=null);
+//        Assert.assertTrue(captorListener.getValue() instanceof AppsflyerIntegration.ConversionListener);
+//        Assert.assertTrue(captorDevKey.getValue() == settings.getString("appsFlyerDevKey"));
+//        Assert.assertTrue(captorContext.getValue() == app.getApplicationContext());
+//        verify(appsFlyerLib).subscribeForDeepLink(AppsflyerIntegration.deepLinkListener);
+
+    @Test
+    public void testAppsflyerIntegration_FACTORYKEY_happyFlow() throws Exception {
+            Assert.assertEquals(AppsflyerIntegration.FACTORY.key(),"AppsFlyer");
+    }
+
+    //need to check params values are null
+//    @Test
+//    public void testAppsflyerIntegration_onActivityCreated_nilFlow() throws Exception {
+//        MockedStatic<AppsFlyerLib> staticAppsFlyerLib = mockStatic(AppsFlyerLib.class);
+//        AppsFlyerLib appsFlyerLib = mock(AppsFlyerLib.class);
+//        staticAppsFlyerLib.when(AppsFlyerLib::getInstance).thenReturn(appsFlyerLib);
+//        AppsflyerIntegration.manualMode=false;
+//        AppsflyerIntegration appsflyerIntegration = mock(AppsflyerIntegration.class);
+//        appsflyerIntegration.onActivityCreated(mock(Activity.class), mock(Bundle.class));
+//        verify(appsFlyerLib).start(any());
+//    }
+
+    @Test
+    public void testAppsflyerIntegration_getUnderlyingInstance_happyFlow() throws Exception {
+        AppsFlyerLib appsFlyerLib = mock(AppsFlyerLib.class);
+        Logger logger = new Logger("test", Analytics.LogLevel.INFO);
+        AppsflyerIntegration appsflyerIntegration = new AppsflyerIntegration(null,logger,appsFlyerLib,null);
+
+        Assert.assertEquals(appsflyerIntegration.getUnderlyingInstance(),appsFlyerLib);
+
+        reset(appsFlyerLib);
+    }
+
+    @Test
+    public void testAppsflyerIntegration_identify_happyFlow() throws Exception {
+        AppsFlyerLib appsFlyerLib = mock(AppsFlyerLib.class);
+        Logger logger = spy(new Logger("test", Analytics.LogLevel.INFO));
+        AppsflyerIntegration appsflyerIntegration = spy(new AppsflyerIntegration(null,logger,appsFlyerLib,null));
```

Comment:
`spy` is redundant

Reply:
The reason I use it is that in the function - identity - the logger calls its function 'verbose' and I want to stub it so I can't just mock it, I must partially mock it.
Maybe I'm wrong..